crypto/hash/hash.go

```diff
@@ -93,6 +94,24 @@ func (hash *Hash) IsEqual(target *Hash) bool {
 	return *hash == *target
 }
 
+// Difficulty returns the leading Zero **bit** count of Hash in binary.
+//  return -1 indicate the Hash pointer is nil
+func (hash *Hash) Difficulty() (difficulty int) {
+	if hash == nil {
+		return -1
+	}
+
+	for i, _ := range *hash {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should omit 2nd value from range; this loop is equivalent to `for i := range ...`